### ISSUE TITLE
Feature request: When comparing folders, add the option to select different comparison type if files match and don't match

### ISSUE THREAD

**Issue by Craftist** [WinMerge/winmerge#885]
### Discussed in https://github.com/WinMerge/winmerge/discussions/884

<div type='discussions-op-text'>

<sup>Originally posted by **Craftist** July 18, 2021</sup>
Hi, so I decided to compare two backup versions of my system that were made with True Image.
So I mounted them as K: (older) and L: (newer), and ran a comparison with Options... > Compare Method: Modified Date and Size method.
I went with this one rather than the default one because otherwise it would take ages to complete.
The problem is, WinMerge showed some files as different while in reality the contents were identical but the date wasn't.
I perfectly understand that this is the expected behavior from the method that I've chosen.

So I suggest adding two options:
1. Additional Compare Methods > When files are identical: 
2. Additional Compare Methods > When files are different: 
Both options include radio buttons with the options just like the original Compare Method, but you may exclude the method that was chosen as the original (remove from the submenu completely or gray it out or something) as that doesn't make sense.
Also there may be options like `(none)` (set by default) in order to utilize the current behavior (without the additional checks).

So how will it work:
WinMerge will compare files with the Compare Method as it does it now, but if the additional compare method(s) is/are chosen, it will also compare against them, and the final verdict will be made after the second check.

Some example situations:
Folders to compare:
```
+---------------------+--------------------+---------+ +---------------------+--------------------+---------+
| Folder/file name    | Modified Date      | Size    | | Folder/file name    | Modified Date      | Size    |
+---------------------+--------------------+---------+ +---------------------+--------------------+---------+
| K:\Folder           |                    |         | | L:\Folder           |                    |         |
|   \Folder\file1.txt | 2021.07.18 3:31 PM | 3.11 KB | |   \Folder\file1.txt | 2021.07.19 3:31 PM | 3.11 KB |
|   \Folder\file2.txt | 2021.07.18 3:32 PM | 3.12 KB | |   \Folder\file2.txt | 2021.07.19 3:32 PM | 5.12 KB |
|   \Folder\file3.txt | 2021.07.18 3:33 PM | 3.13 KB | |   \Folder\file3.txt | 2021.07.18 3:33 PM | 3.13 KB |
|   \Folder\file4.txt | 2021.07.18 3:34 PM | 3.14 KB | |   \Folder\file4.txt | 2021.07.18 3:34 PM | 5.14 KB |
+---------------------+--------------------+---------+ +---------------------+--------------------+---------+
```

Current behavior as WinMerge 2.16.12.0 shows (Method: Modified Date and Size):
```
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
| Folder/file name    | Comparison result   | Left date            | Right date              | Left Size | Right Size |
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
| K:\Folder           |                     |                      |                         |           |            |
|   \Folder\file1.txt | Files are different | 7/18/2021 3:31:30 PM | * 7/19/2021 3:31:30 PM  | 3.11 KB   | 3.11 KB    |
|   \Folder\file2.txt | Files are different | 7/18/2021 3:32:30 PM | * 7/19/2021 3:32:30 PM  | 3.11 KB   | 5.11 KB    |
|   \Folder\file3.txt | Identical           | 7/18/2021 3:33:30 PM |   7/18/2021 3:33:30 PM  | 3.11 KB   | 3.11 KB    |
|   \Folder\file4.txt | Files are different | 7/18/2021 3:34:30 PM |   7/18/2021 3:34:30 PM  | 3.11 KB   | 5.11 KB    |
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
```

Expected behavior with Method: Modified Date and Size, Method if file are different: Full Contents
```
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
| Folder/file name    | Comparison result   | Left date            | Right date              | Left Size | Right Size |
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
| K:\Folder           |                     |                      |                         |           |            |
|   \Folder\file1.txt | Identical           | 7/18/2021 3:31:30 PM | * 7/19/2021 3:31:30 PM  | 3.11 KB   | 3.11 KB    |
|   \Folder\file2.txt | Files are different | 7/18/2021 3:32:30 PM | * 7/19/2021 3:32:30 PM  | 3.11 KB   | * 5.11 KB  |
|   \Folder\file3.txt | Identical           | 7/18/2021 3:33:30 PM |   7/18/2021 3:33:30 PM  | 3.11 KB   | 3.11 KB    |
|   \Folder\file4.txt | Files are different | 7/18/2021 3:34:30 PM |   7/18/2021 3:34:30 PM  | 3.11 KB   | * 5.11 KB  |
+---------------------+---------------------+----------------------+-------------------------+-----------+------------+
```

It checked the contents of only file1, file2 and file4 as those have either different dates or different sizes. Therefore file3 can have different contents but have the same date and size. In order to mitigate that, one still has to use Compare Method = Full Contents.
file1, although newer, contains the same data on both K and L, and therefore the second check summarized them as identical.
file2 remains different, as sizes are different, therefore the contents are also different.
file4 also remains different, even though the dates are the same, but sizes differ.

Hopefully this example did not confuse you.

In summary (also counts as a **TL;DR**): add the ability to second-check identical and different files (with separate options for them). Customize the method for this.</div>

**Comment by sdottaka** [comment]
This can already be achieved.

In the Options window (Compare → Folder), set the Compare method to
“Modified Date and Size”, and add the following to
“Additional comparison conditions”:

`allequal(Content)`

This compares file contents after date and size match.

<img width="683" height="572" alt="Image" src="https://github.com/user-attachments/assets/a3695778-9763-440e-8c0c-d32b93600902" />
